src/wallet.rs

```diff
@@ -247,19 +294,71 @@ pub fn gen_keypair_file(outfile: String) -> Result<String, Box<error::Error>> {
     Ok(serialized)
 }
 
+pub enum WalletRpcRequest {
+    ConfirmTransaction,
+    GetAccountInfo,
+    GetBalance,
+    GetFinality,
+    GetLastId,
+    GetTransactionCount,
+    RequestAirdrop,
+    SendTransaction,
+}
+impl WalletRpcRequest {
+    fn make_rpc_request(
+        &self,
+        rpc_addr: &SocketAddr,
+        id: u64,
+        params: Option<String>,
+    ) -> Result<Value, Box<error::Error>> {
+        let rpc_string = format!("http://{}", rpc_addr.to_string());
```

Comment:
Does `reqwest` support TLS too?   If so it would be nice to not hard-code that `http`, so the wallet can talk to https://api.testnet.solana.com

Reply:
I'll look into it.

Reply:
Yep, it does support TLS. I just need to offer a way to pass in the proxy address to use instead of the network address